crates/re_types_builder/src/codegen/docs/mod.rs

```diff
@@ -40,76 +41,223 @@ impl CodeGenerator for DocsCodeGenerator {
 
         let mut filepaths = BTreeSet::new();
 
-        let components = objects.ordered_objects(Some(ObjectKind::Component));
-        for object in objects.ordered_objects(Some(ObjectKind::Archetype)) {
+        let (mut archetypes, mut components, mut datatypes) = (Vec::new(), Vec::new(), Vec::new());
+        let object_map = &objects.objects;
+        for object in objects.ordered_objects(None) {
```

Comment:
unrelated but, `ordered_objects` actually doesn't do any ordering anymore, just filtering, so it would make more sense to call it `object_if_kind(ObjectKind::Datatype)` or similar, and just do `for object in objects.all()` here or something